tutorials/puppet-in-google-cloud-platform.md

```diff
@@ -0,0 +1,102 @@
+
+## Setting up the virtual machine
+
+1. Create a small(1 shared vCPU + 1.7 GB memory) Compute Engine instance with the OS Ubuntu 16.04 xenial and with 'Allow HTTP traffic' option checked under Firewall section and name is puppet-agent.
+
+2. Create another Compute Engine instance but this time a with 1 vCPU + 3.75 GB memory with the same OS (Ubuntu 16.04 xenial). Keep the default firewall option. No need to check http or https. Name the instance as puppet-master.
+
+## Install and run Puppet Master server
+
+1. SSH into the puppet-master and run the following commands to install puppet into the puppet-master
+
+        wget https://apt.puppetlabs.com/puppetlabs-release-pc1-xenial.deb
+        sudo dpkg -i puppetlabs-release-pc1-xenial.deb
+        sudo apt-get update
+        sudo apt-get install puppetserver
+
+2. Start the server by
+
+        sudo systemctl start puppetserver
+
+3. Make sure puppet server is running by
+
+        sudo systemctl status puppetserver
+
+We should see a line that says "active (running)"
+
+4. Now that we've ensured the server is running, we can configure it to start at boot by
+
+        sudo systemctl enable puppetserver
+
+## Install Puppet Agent and request a certificate from Master
+
+1. Now SSH in to the puppet agent and run the following commands to install puppet into puppet-Agent
+
+        wget https://apt.puppetlabs.com/puppetlabs-release-pc1-xenial.deb
+        sudo dpkg -i puppetlabs-release-pc1-xenial.deb
+        sudo apt-get update
+        sudo apt-get install puppet-agent
+
+2. Before starting Puppet Agent we want to make sure that it knows the Puppet Master address to request the certificate. To do that we will edit the        /etc/hosts file. At the end of the file, specify the Puppet master server as follows:
+
+        Puppet_Master_Compute_Engine_Instance_Internal_IP_Adress      Puppet
+
+3. Now run the following Commands to run the Puppet agent
+
+        sudo systemctl start puppet
+        sudo systemctl enable puppet
+
+## Sign the certificate to establish a communication
+
+1. The first time Puppet runs on an agent node, it sends a certificate signing request to the Puppet master. To list all unsigned certificate requests, run the following command on the Puppet master.
+
+        sudo /opt/puppetlabs/bin/puppet cert list
+
+2. We'll use the --all option to sign certificate:
+        sudo /opt/puppetlabs/bin/puppet cert sign --all
+
+You can also do a single sign by
+        sudo /opt/puppetlabs/bin/puppet cert sign puppet-agent.c.YOUR_PROJECT_ID.internal
+
+## Write a simple webserver module and Manifest that will install Apache2 and write hello world page
+
+1. In the puppet-master, go to folder /etc/puppetlabs/code/environments/production/manifests/ and make a manifest file site.pp as
+
+        node /agent/{
+          include webserver
+        }
+
+2. Now go to modules directory by
```

Comment:
2. Navigate to the `/etc/puppetlabs/code/environments/production/modules` directory and make a new directory by running the following command:

     `sudo mkdir -p webserver/manifests`